scripts/testy_launch.py

```diff
@@ -117,140 +117,155 @@ def snapshot_exists(snapshot_id):
-        user = get_value_from_launch_template(distro, 'User')
+        user = get_tag_value_from_resource(instance_id, "User")
 
-        wait_instance_running(instance_id)
-        wait_instance_status_check_ok(instance_id)
     except Exception as e:
+        print("Failed.", flush=True)
         return {"status": 1, "msg": str(e).strip()}
 
-    return {"status": 0, "user": user, "hostname": hostname}
+    return {"status": 0, "user": user, "hostname": hostname,
+        "instance_id": instance_id, "instance_name": instance_name}
 
-# Launch an AWS instance given a snapshot ID.
-# This function returns a dictionary with a 'status' field that is set to 0 only when the instance
-# has been launched successfully. In that case, the dictionary contains a 'user' and 'host' fields.
-# Upon failure, the 'status' field is not 0 and the 'msg' field contains the error message.
-def testy_launch_snapshot(snapshot_id):
+# Launch an AWS instance from a snapshot ID.
+def launch_from_snapshot(snapshot_id):
 
-    try:
-        if not snapshot_exists(snapshot_id):
-            raise Exit(f"The snapshot {snapshot_id} does not exist.")
+    if not snapshot_exists(snapshot_id):
+        return {"status": 1, "msg": f"The snapshot '{snapshot_id}' does not exist."}
+
+    print("", end=f"\rCreating a testy server from snapshot '{snapshot_id}' ... ", flush=True)
+    hostname = None
+    user = None
 
+    try:
         snapshot_status = get_snapshot_status(snapshot_id)
         if snapshot_status != 'completed':
+            print("Failed.", flush=True)
             raise Exit(f"The snapshot '{snapshot_id}' is incomplete ({snapshot_status}).")
 
-        launch_template_name = get_value_from_snapshot(snapshot_id, 'LaunchTemplateName')
-        image_name = f"{launch_template_name}-{snapshot_id}"
+        image_name = f"image-{snapshot_id}"
         image_id = get_image_id(image_name)
 
         if not image_id:
-            architecture = get_value_from_launch_template(launch_template_name, 'Architecture')
+            architecture = get_tag_value_from_resource(snapshot_id, "Architecture")
             image_id = register_image_from_snapshot(image_name, architecture, snapshot_id)
 
         # Launch an EC2 instance based on a template and an image ID.
+        ltname = get_tag_value_from_resource(snapshot_id, "LaunchTemplateName")
         result = local(f"aws ec2 run-instances \
-            --launch-template LaunchTemplateName={launch_template_name} \
-            --tag-specifications 'ResourceType=instance, Tags=[{{Key=Application,Value=testy}}, \
-                {{Key=LaunchTemplateName,Value={launch_template_name}}}]' \
+            --launch-template LaunchTemplateName={ltname} \
             --image-id {image_id} \
             --query Instances[*].InstanceId \
             --output text", hide=True, warn=True)
         if result.stderr:
             raise Exit(result.stderr)
+
+        print("Success!", flush=True)
```

Comment:
Should this be after the wait_on_status_check? Is the server succesfully created if the wait_on_status_check fails later on?

Reply:
Yes, this is here intentionally. The instance is successfully created in EC2 at this point even if anything afterwards fails. If the `wait_on_status_check` fails, it prints a message asking the user to check the machine on the AWS console. Happy to add more comments and/or output if that would be useful!

Reply:
Thanks for the clarification, that makes sense.